sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java

```diff
@@ -104,6 +109,39 @@ protected void cleanUpResources() {
         }
     }
 
+    @Test
+    public void canCreateAndUpdateVirtualMachineWithUserData() {
+        String userDataForCreate = "N0ZBN0MxRkYtMkNCMC00RUM3LUE1RDctMDY2MUI0RTdDNzY4";
+        String userDataForUpdate = "Njc5MDI3MUItQ0RGRC00RjdELUI5NTEtMTA4QjA2RTNGNDRE";
+
+        // Create
+        VirtualMachine vm = computeManager
+            .virtualMachines()
+            .define(vmName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withNewPrimaryNetwork("10.0.0.0/28")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withoutPrimaryPublicIPAddress()
+            .withPopularWindowsImage(KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2019_DATACENTER_GEN2)
+            .withAdminUsername("Foo12")
+            .withAdminPassword(password())
+            .withNewDataDisk(127)
+            .withSize(VirtualMachineSizeTypes.fromString("Standard_D2a_v4"))
+            .withUserData(userDataForCreate)
+            .create();
+        Response<VirtualMachineInner> response = computeManager.serviceClient().getVirtualMachines()
+            .getByResourceGroupWithResponse(rgName, vmName, InstanceViewTypes.USER_DATA, Context.NONE);
+        Assertions.assertEquals(userDataForCreate, response.getValue().userData());
+
+        // Update
+        computeManager.serviceClient().getVirtualMachines()
```

Comment:
Why use serviceClient method for updating VirtualMachine? I guess there can be:
```java
vm.update()
    .withUserData(userDataForUpdate)
    .apply()
```
right?

Reply:
Using `vm.update().withUserData(userDataForUpdate).apply()` cannot update the `VirtualMachine` service in real time, so use the serviceClient method to updating the `VirtualMachine`.

Reply:
We did some parameter copying from `innerModel` to update parameter.
https://github.com/Azure/azure-sdk-for-java/blob/04f9b83c351f6d817da6c73308dc3a37c64d03db/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java#L2764-L2781

 Probably need to add additional `userData` copy here.

Reply:
It has been fixed in the new version and `userData` can be updated.